### PR TITLE
Fix a false negative for ``duplicate-argument-name``

### LINKED ISSUE #9669: False negatives for `duplicate-argument-name`
### Bug description

Currently only positional-or-keyword and keyword-only arguments are taken into account for this check:

```python

def foo1(_, _): # [duplicate-argument-name]
...

def foo3(_, _=3): # [duplicate-argument-name]
...

def foo4(_, *, _): # [duplicate-argument-name]
...
```

positional-only, *varargs and **kwargs are not considered currently:

```python
def foo5(_, *_):
    """*vararg"""

def foo6(_, /, _):
    """positional-only"""

def foo7(_, **_):
    """**kwargs"""
```

### Configuration

_No response_

### Command used

```shell
pylint a.py
```

### Pylint output

```python
The message is not emitted because of the false negatives.
```

### Expected behavior

E0108: Duplicate argument name

### Pylint version

```shell
pylint 3.3.0-dev0
astroid 3.3.0-dev0
Python 3.12.0
```

### OS / Environment

_No response_

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by mbyrnepr2**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
|    | :sparkles: New feature |
|    | :hammer: Refactoring   |
|    | :scroll: Docs          |

## Description
Fix a false negative for ``duplicate-argument-name`` by including ``positional-only``, ``*args`` and ``**kwargs`` arguments in the check.

Closes #9669

### CODE REVIEW

File: pylint/checkers/base/basic_error_checker.py
Comment (mbyrnepr2) [inline pylint/checkers/base/basic_error_checker.py:148]: This is an incidental change to add quotes to the name of the missing argument in the message output.

File: pylint/checkers/base/basic_error_checker.py
Comment (Pierre-Sassoulas) [inline pylint/checkers/base/basic_error_checker.py:289]:
```suggestion
        for arg in node.args.arguments:
```
Wouldn't it works ?

File: tests/functional/d/duplicate/duplicate_argument_name.py
Comment (Pierre-Sassoulas) [inline tests/functional/d/duplicate/duplicate_argument_name.py:22]:
```suggestion
def foo6(_, /, _, *_, _="a", **_):
```
Underscore is not the most readable variable name to use imo, but good enough. Also, for clarity it's probably clearer if there's something else than an underscore unless really needed to create the message?

If I understand correctly we could have this too ?
```suggestion
def foo6(_, /, _, *a, _="my string", **b):
```

**Review by Pierre-Sassoulas — DISMISSED** [review]
LGTM, some nits.

File: tests/functional/d/duplicate/duplicate_argument_name.py
Comment (mbyrnepr2) [reply tests/functional/d/duplicate/duplicate_argument_name.py:22]: As-is the message is emitted 4 times; so following the suggestion will emit it 2 times instead.
Having said that, I don't disagree it's difficult to read - I went a bit mad with underscore :D

I could also split this into separate functions - just to make the illustration a bit easier to digest?

Comment (mbyrnepr2) [reply tests/functional/d/duplicate/duplicate_argument_name.py]: Let me know if the latest commit is reasonable to improve this.
(also maybe the underscore was originally used to supress the unused-argument message :))

Comment (Pierre-Sassoulas) [reply tests/functional/d/duplicate/duplicate_argument_name.py]: Ho yeah, I misunderstood what it was doing, and now it's clear 😄

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by Pierre-Sassoulas** [comment]
Let's wait for another review, 3.3.0 is not out yet :)

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**
